### DISCUSSION TITLE
$this->halt failed in afterFill() function

### DISCUSSION THREAD

**Discussion by zbadshah** [filamentphp/filament#5327] [Help]
It is working fine with other hooks but this one.

It could be a bug but I am not sure.

**Comment by danharrin** [comment]
What do you want it to do if you halt there?

**Reply by zbadshah** [reply]
Well, in this case, the halt is stopping some operation to be done. Specifically, I test a user id and stop him to proceed further with editing.

**Reply by danharrin** [reply]
Redirect then? Halting just stops the current thing from taking place

**Reply by zbadshah** [reply]
Thanks Dan for responding. Could you please clarify:

1. What you suggested will definitely work but the redirect takes affect after saving, while I want to avoid saving the changes? If I can redirect before the changes, please let me know how. My apologies in advance for being new with Filament as well as Laravel/PHP.

2. Also Dan, the halt function should work as documented for all hooks. Why it works for all except afterFill()?

**Reply by zbadshah** [reply]
Kindly help by answering and clarifying above query. Thanks

**Reply by danharrin** [reply]
```php
protected function afterFill()
{
    $this->redirect('url');
}
```

Halt works everywhere, but you haven't told it what to do when it halts. The filling process should be halted in `afterFill()`, but filling has already taken place. So there is no effect.

**Reply by zbadshah** [reply]
Happy New Year Dan and thanks for reply.
Actually, halt is giving error message in afterFill().

Anyway, redirect is a better option. Thanks again.
